### DISCUSSION TITLE
No Conflicts with Git but Github Shows Conflicts

### DISCUSSION THREAD

**Discussion by rdmccart** [community/community#112811] [New to GitHub]
Hello, I'm new to git and github and I ran into a situation where I made a commit in git and pushed to origin without any conflicts being raised
but when I went to github and createa pull request I found over 20.
I ran the git status command and it returned "nothing to commit, working tree clean."

Here are the step I took:
1) git fetch
2) git checkout main
3) git pull origin main
4) git checkout -b "my branch"
5) Outside of git I then physicall moved my files to my local repo and went back to git.
6) git add .
7) git commit -m "my message"
8) git push origin "my branch"

What did I go wrong here?
Is there a way to tell github to ignore these conflicts and just add my files to dev?
My files don't currently exist in dev so I would expect no conflicts.
Also if someone could tell me in simple terms why git said I had no conflicts but github tells me I have 25 I would be forever in your debt.

![Conflicts with Github not with Git for posting](https://github.com/community/community/assets/23322921/427284ea-0d68-420d-98b4-a7deae7ce091)

**Comment by ghost** [comment]
Greetings @rdmccart ,

It's possible that the conflicts you're encountering on GitHub are due to changes made in the main branch after you created your branch. When you create a new branch (my branch) from main and make changes there, if main (or any other branch that your branch is based on) receives new commits before you merge your changes, conflicts can arise during the pull request.

GitHub cannot simply ignore conflicts because they indicate a discrepancy between the changes you're trying to merge and the current state of the branch you're merging into. However, you can resolve these conflicts locally and then push the changes to your branch (my branch). Once resolved, you can create a new pull request or update the existing one.

**When you run git status, it only checks the status of your local repository against its tracked branches.** If there are no changes locally that conflict with the currently checked-out branch (my branch in this case), it will show "nothing to commit, working tree clean." However, conflicts may arise when you try to merge your changes into another branch (such as main) that has received new changes since you created your branch. Git won't detect these conflicts until you attempt to merge or pull changes.

in summary assume this case : 
There's a repo let's call it **A** and you forked this on your own repo let's call it **B**
there might be multiple forks/PRs on **A** so when the owner merges some changes on **A**
**the **B** repo must be synced with A in terms of upstream changes/commits** otherwise when you submit PRs from **B**/branch-ex1 to **A**/branch-ex2
conflicts may occur ,
there are two options to resolve conflicts :

1. Resolve by your self 
2. Resolve by owner of remote repo

_Please mark this discussion as answered if you already got your answer I hope you got ! :)_
Good luck.

**Comment by rdmccart** [comment]
Hi @KohoolateSen ,
Thanks for the response. I try to follow the instructions here for resolving conflicts using command line but step 3 says run git status and when I do that it says "nothing to commit, working tree clean."
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line

**Comment by jfullstackdev** [comment]
maybe you forgot to pull from GitHub first, so all changes in there is not in your local

here is the answer of Copilot,

---

The issue you're facing is likely due to differences between your local branch and the branch you're trying to merge into on GitHub. When you create a pull request on GitHub, it's trying to merge your changes into the target branch (in this case, `dev`). If `dev` has changed significantly since you last synced with it, there could be conflicts even if there were none when you pushed your branch.

Here's a simplified explanation: Git tracks changes to files, not the files themselves. When you create a pull request, Git tries to apply your changes onto the target branch. If the parts of the files you've changed also have changes in the target branch, Git won't know which changes to keep and which to overwrite, causing a conflict.

To resolve this, you need to sync your branch with the latest changes from `dev` and resolve any conflicts. Here's how you can do it:

1. Fetch the latest changes from the remote repository:

```bash
git fetch origin
```

2. Checkout to your branch:

```bash
git checkout "my branch"
```

3. Merge the latest changes from `dev` into your branch:

```bash
git merge origin/dev
```

At this point, Git will either successfully merge the changes, or tell you that there are conflicts. If there are conflicts, you'll need to resolve them:

1. Open the files with conflicts in your text editor. Git will have marked the conflicting areas with `<<<<<<<`, `=======`, and `>>>>>>>`. Edit the file to resolve the conflicts, keeping the changes you want and removing the conflict markers.

2. After you've resolved all conflicts, stage the resolved files:

```bash
git add <resolved-file>
```

3. Once all conflicts are resolved and the resolved files are staged, commit the merge:

```bash
git commit -m "Resolved merge conflicts"
```

4. Finally, push your branch to the remote repository:

```bash
git push origin "my branch"
```

Now, you should be able to create a pull request on GitHub without conflicts. Remember, it's important to regularly sync your branch with the target branch to minimize conflicts when you create a pull request.

**Reply by ramazan-biyik** [reply]
Github cant evaluate the conflicted file. It is not a consistent issue. Please see the RESOLVE CONFLICTS button. It is a disabled button. In normal cases it is enabled.

**Reply by profPlum** [reply]
You didn't read the question or the answer you provided.

**Reply by IsHacker003** [reply]
Ai slop.

**Comment by siimaus** [comment]
This is actual real issue. 

I check with exact commits github gives about PR and it is not shown locally as conflicted.
On our internal repo in GH:

`gh pr view 629 --json mergeable,mergeStateStatus,baseRefOid,headRefOid`

gives
```json
{
  "baseRefOid": "441f79e6229d162f9466e1df526db81b09946a17",
  "headRefOid": "fb91ee7224c02b679c879b4d888788e8ddfa8616",
  "mergeStateStatus": "DIRTY",
  "mergeable": "CONFLICTING"
}
```
Then on local:

ubuntu@06b2713e3cac:/data/repos/github/experiments/repo$  `git checkout 441f79e6229d162f9466e1df526db81b09946a17`
```log
Previous HEAD position was a2c03e416 Merge pull request #634 from tmill-app/BOS-1382
HEAD is now at 441f79e62 Merge branch 'BOS-3228' of github.com:__redacted__ into develop
```
ubuntu@06b2713e3cac:/data/repos/github/experiments/repo$  `git reset --hard`
```log
HEAD is now at 441f79e62 Merge branch 'BOS-3228' of github.com:__redacted__ into develop
```
ubuntu@06b2713e3cac:/data/repos/github/experiments/repo$ `git merge --no-commit --no-ff fb91ee7224c02b679c879b4d888788e8ddfa8616`

```log
Auto-merging src/API/Controllers/CampaignsController.cs
Auto-merging src/Application/Queries/Handlers/Extensions.cs
Auto-merging src/Infrastructure/EF/DataInitializer.cs
Auto-merging src/Infrastructure/EF/Repositories/CampaignRepository.cs
Auto-merging src/Infrastructure/EF/TradingAccountDbContext.cs
Auto-merging src/Infrastructure/Extensions.cs
Automatic merge went well; stopped before committing as requested
```

But github shows conflicts

<img width="903" height="433" alt="image" src="https://github.com/user-attachments/assets/aff75127-762c-4273-99e0-33f92407c043" />

Hence, it is some internal issue of Github.

And files it shows conflicts for, they do not even appear in local merge.

**Comment by zorelias** [comment]
I'm having the same issue in one of my repos. I have had a few instances now of GH saying there's conflicts, but local merge works fine. I've ensured all local branches are up-to-date with GH and beyond that there's not much else I can do.

In GitHub:
- I opened a PR for `<feature> -> qa`
- GH complains about two files conflicting

On local:
```sh
git checkout <feature>
git pull
git checkout qa
git pull
git merge <feature>
# no conflicts
```

My current workaround is to create a local branch based off of the base branch that's giving me issues in GH, merge my feature, push and then merge that instead of the original PR, but this is creating lots of PR noise and really feels like something that shouldn't be happening in the first place.
